src/shared/database/amendments.ts

```diff
@@ -10,77 +14,163 @@ const log = logger({ name: 'amendments' })
  *
- * @returns The list of amendment names.
+ * @param network - The network being retrieved.
+ * @param url - The Faucet URL of the network.
+ *
+ * @returns Void.
  */
-async function fetchAmendmentNames(): Promise<Map<string, boolean> | null> {
+async function fetchNetworkAmendments(
+  network: string,
+  url: string,
+): Promise<void> {
   try {
-    const response = await axios.get(
-      'https://raw.githubusercontent.com/XRPLF/rippled/develop/src/libxrpl/protocol/Feature.cpp',
-    )
-    const text = response.data
-    const amendmentNames: Map<string, boolean> = new Map()
-    text.split('\n').forEach((line: string) => {
-      const name = ACTIVE_AMENDMENT_REGEX.exec(line)
-      if (name) {
-        amendmentNames.set(name[1], name[0].includes('VoteBehavior::Obsolete'))
-      } else {
-        const name2 = RETIRED_AMENDMENT_REGEX.exec(line)
-        if (name2) {
-          amendmentNames.set(name2[1], true)
-        }
-      }
+    log.info(`Updating amendment info for ${network}...`)
+    const client = new Client(url)
+    await client.connect()
+    const featureResponse: FeatureAllResponse = await client.request({
+      command: 'feature',
     })
-    return amendmentNames
-  } catch (err) {
-    log.error('Error getting amendment names', err)
-    return null
+
+    const features = featureResponse.result.features
+
+    for (const id of Object.keys(features)) {
+      amendmentIDs.set(id, {
+        name: features[id].name,
+        deprecated: RETIRED_AMENDMENTS.includes(features[id].name),
+      })
+    }
+
+    await client.disconnect()
+
+    log.info(`Finished updating amendment info for ${network}...`)
+  } catch (error) {
+    log.error(
+      `Failed to update amendment info for ${network} due to error: ${String(
+        error,
+      )}`,
+    )
   }
 }
 
 /**
- * Extracts Amendment ID from Amendment name inside a buffer.
+ * Fetch an amendment info from a network.
  *
- * @param buffer -- The buffer containing the amendment name.
+ * @param amendment_id - The id of the amendment to fetch.
+ * @param url - The Faucet URL of the network.
  *
- * @returns The amendment ID string.
+ * @returns True if the amendment is found, false otherwise.
  */
-function sha512Half(buffer: Buffer): string {
-  return createHash('sha512')
-    .update(buffer)
-    .digest('hex')
-    .toUpperCase()
-    .slice(0, 64)
+async function fetchSingleAmendment(
+  amendment_id: string,
+  url: string,
+): Promise<boolean> {
+  try {
+    log.info(`Updating amendment info for ${amendment_id}...`)
+    const client = new Client(url)
+    await client.connect()
+    const featureResponse: FeatureOneResponse | ErrorResponse =
+      await client.request({
+        command: 'feature',
+        feature: amendment_id,
+      })
+
+    if ('result' in featureResponse) {
+      const feature = featureResponse.result[amendment_id]
+      amendmentIDs.set(amendment_id, {
+        name: feature.name,
```

Comment:
Is there more that should be stored?

Reply:
There are only these fields in `feature`, of which only name is useful for us.

```
enabled | Boolean | Whether this amendment is currently enabled in the latest ledger.

name | String | (May be omitted) The human-readable name for this amendment, if known.

supported | Boolean | Whether the server knows how to apply this amendment. If this field is set to false (the server does not know how to apply this amendment) and enabled is set to true (this amendment is enabled in the latest ledger), this amendment may cause your server to be amendment blocked.
```